.github/workflows/release.yml

```
@@ -0,0 +1,145 @@
+        description: 'Next development version (e.g., 0.1.1-SNAPSHOT)'
+        required: true
+      dryRun:
+        description: 'Dry run (true or false)'
+        required: false
+        default: false
+        type: boolean
+
+# Required permissions for pushing to main branch and creating tags, issues and discussions
+permissions:
+  contents: write
+  packages: write
+  issues: write
+  discussions: write
+
+jobs:
+  release:
+    name: Release version ${{ inputs.releaseVersion }}
+
+    # Only run in the original repository, not forks
+    if: github.repository == 'dremio/iceberg-auth-manager'
+    runs-on: ubuntu-latest
+    
+    steps:
+
+      - name: Validate inputs
+        run: |
+          if [[ "${{ github.event.inputs.releaseVersion }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?$ && ! "${{ github.event.inputs.releaseVersion }}" =~ -SNAPSHOT$ ]]; then
+            echo "Valid release version: ${{ github.event.inputs.releaseVersion }}"
+          else
+            echo "Invalid release version format. Expected format: x.y.z or x.y.z-qualifier"
+            exit 1
+          fi
+          if [[ "${{ github.event.inputs.nextVersion }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+-SNAPSHOT$ ]]; then
+            echo "Valid next development version: ${{ github.event.inputs.nextVersion }}"
+          else
+            echo "Invalid next development version format. Expected format: x.y.z-SNAPSHOT"
+            exit 1
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0 # Fetch all history for all branches and tags
+          ref: main # Checkout the main branch
+
+      - name: Verify current version
+        run: |
+          if [[ ! -f version.txt ]]; then
+            echo "version.txt file not found!"
+            exit 1
+          fi
+          current_version=$(cat version.txt)
+          if [[ ! "$current_version" =~ -SNAPSHOT$ ]]; then
+              echo "Current version is not a SNAPSHOT version: $current_version"
+              exit 1
+          fi
+
+      - name: Configure Git
+        run: |
+          git config --global user.email "authmgr-release-workflow-noreply@dremio.com"
+          git config --global user.name "AuthManager Release Workflow [bot]"
+
+      - name: Update version for release
+        run: |
+          echo "${{ github.event.inputs.releaseVersion }}" > version.txt
+          git add version.txt
+          git commit -a -m "chore(release): release version ${{ github.event.inputs.releaseVersion }}"
+          git tag -f -a -m "chore(release): tag version ${{ github.event.inputs.releaseVersion }}" "authmgr-${{ github.event.inputs.releaseVersion }}"
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 21
+
+      - name: Build with Gradle
+        run: ./gradlew clean test publish assemble -x intTest --no-build-cache
```

Nit: `clean` and `--no-build-cache` is superfluous. The workspace is "fresh" and a Gradle cache's not restored.

```suggestion
        run: ./gradlew test publish assemble -x intTest
```

Not sure whether `test` is needed. `assemble` should be implicit to `publish`

---

Indeed publish implies assemble, but not test. I think I'll go with `./gradlew build publish -x intTest`.